Base.rte/Activities/BunkerBreach.lua

```diff
@@ -2,213 +2,263 @@
-	for _, loadout in pairs(self.loadouts) do
-		for i = 1, 10 do
-			if SceneMan.Scene:HasArea(loadout .. i) then
-				local guard = self:CreateInfantry(self.defenderTeam, loadout);
-				if guard then
-					guard.Pos = SceneMan.Scene:GetArea(loadout .. i):GetCenterPoint();
-					MovableMan:AddActor(guard);
+function BunkerBreach:SetupDefenderActors()
+	local techID = PresetMan:GetModuleID(self:GetTeamTech(self.defenderTeam));
+	local crabToHumanSpawnRatio = self:GetCrabToHumanSpawnRatio(techID);
+
+	for _, loadoutName in pairs({"Light", "Heavy", "Sniper", "Engineer", "Mecha", "Turret"}) do
+		if SceneMan.Scene:HasArea(loadoutName .. " Defenders") then
+			local defenderArea = SceneMan.Scene:GetOptionalArea(loadoutName .. " Defenders");
+			if defenderArea ~= nil then	
+				for defenderBox in defenderArea.Boxes do
+					local guard;
+					if loadoutName == "Mecha" or loadoutName == "Turret" then
+						guard = crabToHumanSpawnRatio > 0 and self:CreateCrab(techID, loadoutName == "Turret") or self:CreateInfantry(techID, "Heavy");
+					else
+						guard = self:CreateInfantry(techID, loadoutName);
+					end
+					if guard then
+						guard.Pos = defenderBox.Center;
+						guard.Team = self.defenderTeam;
+						guard.AIMode = Actor.AIMODE_SENTRY;
+						if loadoutName == "Engineer" then
+							guard.AIMode = Actor.AIMODE_GOLDDIG;
+						end
+						MovableMan:AddActor(guard);
+					end
 				end
-			else
-				break;
 			end
 		end
 	end
+	for actor in MovableMan.AddedActors do
+		if actor.Team ~= self.defenderTeam and not actor:IsInGroup("Brains") then
+			MovableMan:ChangeActorTeam(actor, self.defenderTeam);
+		end
+	end
+end
+
+function BunkerBreach:SetupDefenderInternalReinforcementAreas()
+	if self.AI.isDefenderTeam then
+		local internalReinforcementsArea = SceneMan.Scene:GetOptionalArea("Internal Reinforcements");
+		if internalReinforcementsArea ~= nil then
+			self.AI.internalReinforcementsDoorParticle = CreateMOSRotating("Animated Background Door", "Base.rte");
+			self.AI.internalReinforcementPositions = {};
+			for internalReinforcementsBox in internalReinforcementsArea.Boxes do
+				local backgroundDoor = CreateTerrainObject("Module Back Middle E", "Base.rte");
+				backgroundDoor.Pos = SceneMan:SnapPosition(internalReinforcementsBox.Corner, true);
+				self.AI.internalReinforcementPositions[#self.AI.internalReinforcementPositions + 1] = backgroundDoor.Pos + Vector(24, 24);
+				SceneMan:AddSceneObject(backgroundDoor);
+			end
+		end
+	end
+end
 
+function BunkerBreach:SetupFogOfWar()
 	if self:GetFogOfWarEnabled() then
 		SceneMan:MakeAllUnseen(Vector(24, 24), self.attackerTeam);
 		SceneMan:MakeAllUnseen(Vector(24, 24), self.defenderTeam);
-		if self.CPUTeam ~= -1 then
-			SceneMan:MakeAllUnseen(Vector(70, 70), self.CPUTeam);
-			--Assume that the AI has scouted the terrain
-			for x = 0, SceneMan.SceneWidth - 1, 65 do
-				SceneMan:CastSeeRay(self.CPUTeam, Vector(x, 0), Vector(0, SceneMan.SceneHeight), Vector(), 1, 9);
-			end
+		
```

Comment:
I think some of these comments have value. WIthout them, I have no idea why it's doing any of this CastSeeRay stuff.